docs/training/apis-howto/api-pytorch-ug.rst

```diff
@@ -638,7 +636,271 @@ Take the following code for example.
+tests around your model code.
+
+.. code:: python
+
+    trainer.fit(
+                 max_length=pytorch.Epoch(1),
+                 checkpoint_period=pytorch.Batch(100),
+                 validation_period=pytorch.Batch(100),
+   +             test_mode=True
+             )
+
+This replaces the legacy test mode codepath, which supports this functionality for trials not using
+Trainer API:
+
+.. code:: bash
+
+   det e create det.yaml . --local --test
+
+Prepare Your Training Code for Deploying to a Determined Cluster
+================================================================
+
+Once you are satisfied with the results of training the model locally, you submit the code to a
+cluster. This example allows for distributed training locally and on cluster without having to make
+code changes.
+
+Example workflow of frequent iterations between local debugging and cluster deployment:
+
+.. code:: diff
+
+    def main():
+   +   local = det.get_cluster_info() is None
+   +   if local:
+   +       # Local: configure local distributed training.
+   +       dist.init_process_group(backend="gloo|nccl")
+   +       os.environ["USE_TORCH_DISTRIBUTED"] = "true"
+   +       distributed_context = core.DistributedContext.from_torch_distributed()
+   +       latest_checkpoint = None
+   +   else:
+   +       # On-cluster: Determined will automatically detect distributed context.
+   +       distributed_context = None
+   +       # On-cluster: configure the latest checkpoint for pause/resume training functionality.
+   +       latest_checkpoint = det.get_cluster_info().latest_checkpoint
+
+   +     with det.pytorch.init(
+   +       distributed=distributed_context
+         ) as train_context:
+             trial = MNistTrial(train_context)
+             trainer = det.pytorch.Trainer(trial, train_context)
+             trainer.fit(
+                 max_length=pytorch.Epoch(1),
+                 checkpoint_period=pytorch.Batch(100),
+                 validation_period=pytorch.Batch(100),
+   +             latest_checkpoint=latest_checkpoint,
+             )
+
+To run Trainer API solely on-cluster, the code is much simpler:
+
+.. code:: python
+
+   def main():
+       with det.pytorch.init() as train_context:
+           trial_inst = model.MNistTrial(train_context)
+           trainer = det.pytorch.Trainer(trial_inst, train_context)
+           trainer.fit(
+               checkpoint_period=pytorch.Batch(100),
+               validation_period=pytorch.Batch(100),
+               latest_checkpoint=det.get_cluster_info().latest_checkpoint,
+           )
+
+Submit Your Trial for Training on Cluster
+=========================================
+
+To run your experiment on cluster, you'll need to create an experiment configuration (YAML) file.
+Your experiment configuration file must contain searcher configuration and entrypoint.
+
+.. note::
+
+   ``hyperparameters.global_batch_size`` is required if ``max_length`` is configured in records
```

Comment:
I recommend deleting this note.

My reasoning is: this is an overly specific detail that is being over-emphasized (via a `note` admonition) in otherwise pretty high-level docs.

I think this information is much better suited to a detail warning message for the users that encounter it.

But at a minimum, demote it from `note` admonition to plain text.  The `note`s that we allow will be among the first things that readers will read when the skim, because it's a visual indicator that "this is really important".  But this particular information is only even relevant for a probably small subset of users.  Don't waste user attention on this information.